src/main/java/fr/sncf/osrd/simulation/IntegrationStep.java

```diff
@@ -0,0 +1,126 @@
+        var action = integrator.actionToTargetSpeed(directive, rollingStock, directionSign);
+        return computeNextStep(
+                integrator,
+                action,
+                distanceLeft,
+                directionSign
+        );
+    }
+
+    /** Computes the next step of the integration method, based on location, speed, and a specified action. */
+    public static IntegrationStep computeNextStepFromAction(TrainPositionTracker currentLocation,
+                                                            double currentSpeed,
+                                                            Action action,
+                                                            RollingStock rollingStock,
+                                                            double timeStep,
+                                                            double end,
+                                                            int directionSign) {
+
+        var integrator = new TrainPhysicsIntegrator(
+                timeStep,
+                rollingStock,
+                currentSpeed,
+                currentLocation.meanTrainGrade());
+        var distanceLeft = end - currentLocation.getPathPosition();
+        return computeNextStep(
+                integrator,
+                action,
+                distanceLeft,
+                directionSign);
+    }
+
+    /** Computes the next step of the integration method, based on location, speed, and a set of controllers. */
+    public static IntegrationStep computeNextStepFromControllers(TrainPositionTracker currentLocation,
+                                                                 double currentSpeed,
+                                                                 Set<SpeedController> controllers,
+                                                                 RollingStock rollingStock,
+                                                                 double timeStep,
+                                                                 double end,
+                                                                 int stopIndex,
+                                                                 int directionSign) {
+        var integrator = new TrainPhysicsIntegrator(
+                timeStep,
+                rollingStock,
+                currentSpeed,
+                currentLocation.meanTrainGrade());
+        var nextPosition = currentLocation.getPathPosition() + currentSpeed * timeStep;
+        final var finalNextPosition = min(nextPosition, end);
+        var directive = SpeedController.getDirective(controllers, finalNextPosition, stopIndex);
+        var action = integrator.actionToTargetSpeed(directive, rollingStock);
+        var distanceLeft = end - finalNextPosition;
+        return computeNextStep(
+                integrator,
+                action,
+                distanceLeft,
+                directionSign
+        );
+    }
```

Comment:
I'd rather have this stuff somewhere in TrainPhysicsIntegrator, inside a region:

```java
// region NEXT_STEP_HELPERS
        public static IntegrationStep computeNextStep(TrainPhysicsIntegrator integrator, ...)
        ....
// endregion
```